indexer/indexer_tests/categories_test.cpp

```diff
@@ -87,10 +87,7 @@ UNIT_TEST(LoadCategories)
 {
   classificator::Load();
 
-  CategoriesHolder h;
-  istringstream buffer(TEST_STRING);
-  h.LoadFromStream(buffer);
-
+  CategoriesHolder h(new MemReader(TEST_STRING, strlen(TEST_STRING)));
```

Comment:
Юра, я попробовал, сходу не получилось: нужно рефакторить множество кода, принимающего указатели. Предлагаю сделать это в отдельном реквесте.

> On 16.2.2016, at 16.17, ygorshenin <notifications@github.com> wrote:
> 
> In indexer/indexer_tests/categories_test.cpp:
> 
> > @@ -87,10 +87,7 @@ UNIT_TEST(LoadCategories)
> >  {
> >    classificator::Load();
> >  
> > -  CategoriesHolder h;
> > -  istringstream buffer(TEST_STRING);
> > -  h.LoadFromStream(buffer);
> > -
> > +  CategoriesHolder h(new MemReader(TEST_STRING, strlen(TEST_STRING)));
> OK, why not to change CategoriesHolder ctor from:
> 
> CategoriesHolder(Reader * p)
> to
> 
> CategoriesHolder(unique_ptr<Reader> p)
> ?
> It would be more readable and all questions about Reader lifetime will disappear.
> 
> —
> Reply to this email directly or view it on GitHub.
>